lib/common/ipc_client.c

```diff
@@ -305,6 +305,43 @@ call_api_dispatch(pcmk_ipc_api_t *api, xmlNode *message)
     return false;
 }
 
+/* Do the hard work of dispatch_ipc_data.  This is split out into its own
+ * function so it can be shared with pcmk__send_ipc_request.  The return
+ * value is as follows:
+ *
+ * -1: There was an error.
+ *  0: There are no more messages expected from the server.  Quit reading.
+ *  1: There are more messages expected from the server.  Keep reading.
+ */
+static int
+do_dispatch_ipc_data(const char *buffer, pcmk_ipc_api_t *api)
```

Comment:
I'm not a fan of "do_*" functions -- let's call this dispatch_ipc_data(), and rename the existing one to something like dispatch_ipc_source_data(). Also I'd prefer a standard return code even if we only use 3 possibilities.

pcmk_dispatch_ipc() can use this function directly

Reply:
Fixed, though i'm not thrilled with the choice of return codes.